pkg/deployment/depresource/start.go

```diff
@@ -0,0 +1,164 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package depresource
+
+import (
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/client/deployments"
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/ecctl/pkg/deployment"
+	"github.com/elastic/ecctl/pkg/deployment/deputil"
+	"github.com/elastic/ecctl/pkg/util"
+)
+
+// StartParams is consumed by start.
+type StartParams struct {
+	*api.API
+	DeploymentID string
+	Type         string
+	RefID        string
+	All          bool
+}
+
+// Validate ensures the parameters are usable by Start.
+func (params StartParams) Validate() error {
+	var merr = new(multierror.Error)
+
+	if params.API == nil {
+		merr = multierror.Append(merr, util.ErrAPIReq)
+	}
+
+	if len(params.DeploymentID) != 32 {
+		merr = multierror.Append(merr, deputil.NewInvalidDeploymentIDError(params.DeploymentID))
+	}
+
+	if params.Type == "" {
+		merr = multierror.Append(merr, errors.New("deployment resource type cannot be empty"))
+	}
+
+	return merr.ErrorOrNil()
+}
+
+func (params *StartParams) fillDefaults() error {
+	if params.RefID == "" {
+		refID, err := deployment.GetTypeRefID(deployment.GetResourceParams{
+			GetParams: deployment.GetParams{
+				API:          params.API,
+				DeploymentID: params.DeploymentID,
+			},
+			Type: params.Type,
+		})
+		if err != nil {
+			return err
+		}
+
+		params.RefID = refID
+	}
+
+	return nil
+}
+
+// StartInstancesParams is consumed by StartInstances.
+type StartInstancesParams struct {
+	StartParams
+	IgnoreMissing *bool
+	InstanceIDs   []string
+}
+
+// Validate ensures the parameters are usable by StartInstances.
+func (params StartInstancesParams) Validate() error {
+	var merr = new(multierror.Error)
+
+	if len(params.InstanceIDs) == 0 {
```

Comment:
Is this valid if `all` flag is used?

Reply:
That code isn't touched when the `--all` flag is set https://github.com/elastic/ecctl/pull/89/files#diff-e9b68f402a2f2dd49799010fb1b725f1R78-R79 :)